### PR TITLE
Took out streaming from create_run, create_and_process_run, and submit_tool_outputs_to_run and declare separate functions for streaming

### PR DESCRIPTION
**Pull request by howieleung**
…t_tool_outputs_to_run and declare separate functions for streaming

# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ai/azure-ai-client/azure/ai/client/operations/_patch.py [hunk 1]
```diff
@@ -895,10 +868,402 @@ def create_and_process_run(
...
+        :paramtype event_handler: ~azure.ai.client.models.AgentEventHandler
+        :keyword sleep_interval: The time in seconds to wait between polling the service for run status.
+            Default value is 1.
+        :paramtype sleep_interval: int
+        :return: AgentRunStream.  AgentRunStream is compatible with Iterable and supports streaming.
+        :rtype: ~azure.ai.client.models.AgentRunStream
+        :raises ~azure.core.exceptions.HttpResponseError:
+        """
+        # Create and initiate the run with additional parameters
+        run = self.create_stream(
```
Comment (jhakulin) [inline sdk/ai/azure-ai-client/azure/ai/client/operations/_patch.py:1241]: We may not need this method unless we provide some functionality that differs from create_stream, right?

File: sdk/ai/azure-ai-client/samples/agents/async_samples/sample_agents_stream_eventhandler_with_toolset_async.py [hunk 2]
```diff
@@ -141,10 +140,9 @@ async def main():
         message = await ai_client.agents.create_message(thread_id=thread.id, role="user", content="Hello, send an email with the datetime and weather information in New York? Also let me know the details")
         print(f"Created message, message ID {message.id}")
 
-        async with await ai_client.agents.create_and_process_run(
+        async with await ai_client.agents.create_and_process_stream(
```
Comment (jhakulin) [inline sdk/ai/azure-ai-client/samples/agents/async_samples/sample_agents_stream_eventhandler_with_toolset_async.py:143]: we could use only create_stream

File: sdk/ai/azure-ai-client/azure/ai/client/operations/_patch.py (hunk 1 above)
Comment (jhakulin) [reply sdk/ai/azure-ai-client/azure/ai/client/operations/_patch.py:1241]: So let's add it only when we see there is a need.

**Review by jhakulin — CHANGES_REQUESTED** [review]
Added comments

**Review by jhakulin — APPROVED** [review]

File: sdk/ai/azure-ai-client/samples/agents/async_samples/sample_agents_stream_eventhandler_with_toolset_async.py (hunk 2 above)
Comment (howieleung) [reply sdk/ai/azure-ai-client/samples/agents/async_samples/sample_agents_stream_eventhandler_with_toolset_async.py]: Right.   fixed.   Good catch

**Pull request merged**